packages/integration-sdk-runtime/src/storage/FileSystemGraphObjectStore/FileSystemGraphObjectStore.ts

```diff
@@ -98,6 +102,56 @@ function integrationStepsToGraphObjectIndexMetadataMap(
   return stepIdToGraphObjectIndexMetadataMap;
 }
 
+/**
+ * When entities are first ignested, they are stored in a local in-memory
+ * graph object store.
+ */
+type GraphObjectLocationInMemory = {
+  location: 'inMemory';
+};
+
+/**
+ * When entities are flushed from the local in-memory graph obejct store,
+ * they have two options: onDisk or nonIndexable.
+ *
+ * nonIndexable entities are explicitly not required to be stored on disk,
+ * and will not be searchable after they've been flushed from the local
+ * in-memory graph object store.
+ */
+type GraphObjectLocationNonIndexable = {
+  location: 'nonIndexable';
+};
+
+/**
+ * Most entities flushed from the local in-memory graph object store are
+ * flushed to disk.
+ *
+ * These entities can be looked up by their file path and index.
+ */
+type GraphObjectLocationOnDisk = {
+  location: 'onDisk';
+  graphDataPath: string;
+};
+
+/**
+ * The FileSystemGraphObjectStore will keep track of the current location of entities:
+ *
+ * Entities that have yet to be flushed from the InMemoryGraphObjectStore have a location of
+ *   { location: 'inMemory' }
+ *
+ * Entities that have been flushed to disk have a location of
+ *   { location: 'onDisk', graphDataPath: '/graph/step-id/filename.json' }
+ *
+ * Finally, entities that have been flushed from the InMemoryGraphObjectStore, but that
+ * have `indexMetadata.enabled = false`, are explicitly not findable. These have a
+ * location of
+ *   { location: 'nonIndexable' }
+ */
```

Comment:
I wanted to design this to be a holistic view into where entities exist in this system, including whether they are `inMemory` _or_ `nonIndexable`. However, this map will be larger than it needs to be for that purpose. An alternative is to _only_ maintain a map of entities on disk (since finding them in memory is trivial, and we may not care if they have not been indexed on disk).

Would love to know all your thoughts on this.

Reply:
I agree that we do not need to maintain a mapping of things in memory or not indexed or anywhere else. I'd like to see the FileSystemGraphObjectStore encapsulate its management of an index of its data; nothing outside the object should know what it's doing to be fast as what it does. The code that uses an in memory store and flushes to disk when that gets full could be a `BufferedGraphObjectStore` that writes to an in memory store and flushes to a file system store, but provides a single interface for `findEntity`.